thrift/thrift.bzl

```diff
@@ -0,0 +1,62 @@
+"""Rules for organizing thrift files."""
+
+load("//scala:scala.bzl", "implicit_deps")
+
+_thrift_filetype = FileType([".thrift"])
+
+#TODO some unanswered questions...
+# - should the tar for a given target include the deps? I believe the answer is no (they should be transitive)
+# - should "building" this tar trigger the builds of the dependent tars? I believe the answer should be yes...not sure how to trigger that?
+#   hmm, on the other side, though... perhaps the answer is no, and it should only be triggered when it is actually used/needed? yes, that
+#   seems more in line with bazel philosophy of "only build what you need when you need it"
+def _thrift_library_impl(ctx):
+  #TODO how do we want to archive this? could just do another jar to be consistent with everything else...
+  cmd = "tar -c $@ > {out}".format(out=ctx.outputs.libarchive.path)
```

Comment:
yes. exactly. We want the output to be a function of the content of the inputs (and not when it was run). We do take steps to ensure this. That way, the hash function used to check if we need to rebuild works.